cub/grid/grid_even_share.cuh

```diff
@@ -122,17 +122,18 @@ public:
      * \brief Dispatch initializer. To be called prior prior to kernel launch.
      */
     __host__ __device__ __forceinline__ void DispatchInit(
-        OffsetT num_items,          ///< Total number of input items
+        OffsetT num_items_,          ///< Total number of input items
         int     max_grid_size,      ///< Maximum grid size allowable (actual grid size may be less if not warranted by the the number of input items)
         int     tile_items)         ///< Number of data items per input tile
     {
-        this->block_offset          = num_items;    // Initialize past-the-end
-        this->block_end             = num_items;    // Initialize past-the-end
-        this->num_items             = num_items;
-        this->total_tiles           = (num_items + tile_items - 1) / tile_items;
-        this->grid_size             = CUB_MIN(total_tiles, max_grid_size);
+        this->block_offset          = num_items_;    // Initialize past-the-end
+        this->block_end             = num_items_;    // Initialize past-the-end
+        this->num_items             = num_items_;
+        this->total_tiles           = (num_items_ + tile_items - 1) / tile_items;
+        this->grid_size             = CUB_MIN(static_cast<int>(total_tiles), max_grid_size);
```

Comment:
This could overflow, because `total_tiles` is `OffsetT` which could be 64bit.

Reply:
This may be related to #221.

Reply:
I think I'd prefer us to either make sure all the static casts don't overflow, or move them to a separate PR.

Reply:
Fixed this and similar issues in the thrust/cub kernel dispatch code with a new `cub::DivideAndRoundUp` method that avoids these `(n + d - 1) / d` overflows.